src/Common/SystemLogBase.cpp

```diff
@@ -27,12 +27,15 @@
 #include <Common/logger_useful.h>
 #include <base/scope_guard.h>
 
+#define LOGICAL_IF_THEN(A, B) (!(A) || !!(B))
```

Comment:
Move to `base/defines.h` ?

Reply:
Ok, I don't mind.

Reply:
I'm not sure adding this to defines.h, which is included everywhere and will force recompiling everything, is a good idea. It's a macro only used once, so maybe we could even not use a macro.

Reply:
It is better to be a function in general.
For example if we do 
```
#define LOGICAL_IFF(A, B) (LOGICAL_IF_THEN(A, B) && LOGICAL_IF_THEN(B, A))
```
then `A` and `B` could be calculated twice.

But I wanted to leave it as simple as possible. Beside `LOGICAL_IFF` does not have such issue.